src/libaudqt/fileopener.cc

```diff
@@ -39,14 +42,65 @@ static void import_playlist(Playlist playlist, const String & filename)
     playlist.insert_entry(0, filename, Tuple(), false);
 }
 
-static void export_playlist(Playlist playlist, const String & filename)
+static void export_playlist(Playlist playlist, const String & filename,
```

Comment:
I think it would be slightly cleaner if you were to:
1. Make the existing `import/export_playlist()` functions take `const char *`:
```
static void import_playlist(Playlist playlist, const char * filename)
static void export_playlist(Playlist playlist, const char * filename)
```
2. Then add a new overload for `export_playlist()`:
```
static void export_playlist(Playlist playlist, const char * filename,
                            const char * default_ext)
{
    if (uri_get_extension(filename))
        export_playlist(playlist, filename);
    else if (default_ext)
        export_playlist(playlist, str_concat({filename, ".", default_ext}));
    else
        aud_ui_show_error(_("Please type a filename extension or select a "
                            "format from the drop-down list."));
}
```
It avoids the need for `str_copy()` and simplifies the logic slightly.

Reply:
I thought that `default_ext` would be `nullptr` since `QRegularExpression.match().captured()` is supposed to return a "null" QString (which is distinct from an "empty" QString) is there was no match.

Feel free to push a fix as you see fit. I'm curious what I'm missing here though.

Reply:
`QRegularExpression.match().captured()` behaves exactly like that and returns a "null" QString in this case.
`QString.isNull()` returns true.

However this is not working as expected, the parameter still becomes an empty string.

```c++
export_playlist(..., extension.isNull() ? nullptr : extension.toUtf8());
```

Adding `.constData()` seems to help, but I don't know why.

```c++
export_playlist(..., extension.isNull() ? nullptr : extension.toUtf8().constData());
```

Omitting the null check results in an empty string again. Very odd... Does that make sense to you?
I guess adjusting the `else if` as suggested is the simplest way to fix this.

Reply:
Ah I found this: "All functions except [isNull](https://doc.qt.io/qt-6/qstring.html#isNull)() treat null strings the same as empty strings. For example, [toUtf8](https://doc.qt.io/qt-6/qstring.html#toUtf8)().[constData](https://doc.qt.io/qt-6/qstring.html#constData)() returns a valid pointer (not nullptr) to a '\0' character for a null string."

`default_ext && default_ext[0]` sounds like the way to go 👍